geopyspark/geotrellis/layer.py

```diff
@@ -1282,6 +1306,30 @@ def convert_data_type(self, new_type, no_data_value=None):
             return TiledRasterLayer(self.layer_type,
                                     self.srdd.convertDataType(new_type))
 
+    def with_no_data(self, no_data_value):
```

Comment:
The reason I used that naming convention is because that's what is used throughout the rest of the library. Changing the name here would mean changing names of other parameters/values throughout the library. Which might be fine, but it's something we should consider.